types.go

```diff
@@ -58,6 +71,49 @@ func (source *Source) MetadataWithAdditionalTags(tags []string) []MetadataField
 	}
 }
 
+func (source *Source) AuthenticateToECR() bool {
+	os.Setenv("AWS_ACCESS_KEY_ID", source.AwsAccessKeyId)
+	os.Setenv("AWS_SECRET_ACCESS_KEY", source.AwsSecretAccessKey)
+	os.Setenv("AWS_REGION", source.AwsRegion)
```

Comment:
nitpick: Can we do this without mutating global env state? I believe the AWS SDK can also take these as parameters somehow.